R/thumbnail.R

```diff
@@ -0,0 +1,327 @@
+#' Get content item thumbnail
+#'
+#' Download the thumbnail for a content item on Connect to a file on your
+#' computer.
+#'
+#' @param content A content item.
+#' @param path Optional. A path to a file used to write the thumbnail image. If
+#' no path is provided, a temporary file with the correct file extension is
+#' created. If the content item does not have a thumbnail, returns `NA`.
+#' 
+#' @returns The path to the image file.
+#' 
+#' @examples
+#' \dontrun{
+#' client <- connect()
+#' item <- content_item(client, "8f37d6e0-3395-4a2c-aa6a-d7f2fe1babd0")
+#' thumbnail <- get_thumbnail(item)
+#' }
+#'
+#' @family thumbnail functions
+#' @family content functions
+#' @export
+get_thumbnail <- function(content, path = NULL) {
+  validate_R6_class(content, "Content")
+  guid <- content$content$guid
+  con <- content$connect
+
+  # Connect 2024.09.0 introduced public endpoints for content thumbnails. We
+  # prefer those, falling back to the unversioned endpoints if unavailable.
+  res <- con$GET(
+    url = con$server_url("content", guid, "__thumbnail__"),
+    parser = NULL
+  )
+  if (httr::status_code(res) == 404) {
+    res <- con$GET(
+      unversioned_url("applications", guid, "image"),
+      parser = NULL
+    )
+  }
+  con$raise_error(res)
+
+  if (httr::status_code(res) == 204) {
+    return(NA_character_)
+  }
+
+  # Guess file extension
+  if (is.null(path)) {
+    ct <- httr::headers(res)$`content-type`
+    if (grepl("^image/", ct)) {
+      # Just strip off 'image/'
```

Comment:
This comment is inaccurate; this uses mime::mimemap and does not strip an `image/` prefix.

What about using `mime::mimemap` to qualify the content type without looking for an `image/` prefix? Fall back to PNG when it is not a known type.

```r
ext <- if (any(mime::mimemap == ct)) {
    names(mime::mimemap[mime::mimemap == ct])[1]
} else {
    "png"
}
```

Reply:
Yeah, definitely more elegant.